subprojects/docs/src/samples/build-organization/sharing-convention-plugins-with-build-logic/kotlin/buildSrc/build.gradle.kts

```diff
@@ -0,0 +1,35 @@
+// tag::plugins[]
+plugins {
+    id("java-gradle-plugin")
+    id("myproject.java-conventions")
+}
+// end::plugins[]
+
+repositories {
+    jcenter()
+}
+
+dependencies {
+    testImplementation("junit:junit:4.13")
+}
+
+gradlePlugin {
```

Comment:
I think adding this "binary" plugin is making this look unnecessary complex.

I am also not sure if we would recommend users to do binary plugins if they do not publish. In our build we moved almost everything to pre-compiled script plugins. It's also less boilerplate.

I also think all the "functionalTest" stuff below is only distracting from what this sample should be about.

What I would do is turn this _Greeting Plugin_ into a script plugin itself and just have a `GreetTask` implemented in Java.